subsys/storage/flash_map/flash_map.c

```diff
@@ -226,18 +234,26 @@ int flash_area_erase(const struct flash_area *fa, off_t off, size_t len)
 
 	flash_dev = device_get_binding(fa->fa_dev_name);
 
+	if (!flash_dev) {
+		return -ENODEV;
+	}
+
 	rc = flash_erase(flash_dev, fa->fa_off + off, len);
 
 	return rc;
 }
 
 uint8_t flash_area_align(const struct flash_area *fa)
 {
-	const struct device *dev;
+	const struct device *flash_dev;
 
-	dev = device_get_binding(fa->fa_dev_name);
+	flash_dev = device_get_binding(fa->fa_dev_name);
+
+	if (!flash_dev) {
+		return -ENODEV;
+	}
```

Comment:
You can't return an error code for this routine (`uint8_t`)

Reply:
done